io-engine/src/lvm/vg_pool.rs

```diff
@@ -0,0 +1,294 @@
+use serde::Deserialize;
+
+use crate::{
+    bdev::PtplFileOps,
+    lvm::property::Property,
+    pool_backend::PoolArgs,
+};
+
+use super::{
+    cli::{de, CmnQueryArgs, LvmCmd},
+    error::Error,
+};
+
+/// VG query arguments, allowing filtering via --select.
+/// It's essentially a new-type wrapper over the common arguments
+/// which can't be used to build the query because... it's common
+/// between VG and LV..
+#[derive(Default, Debug)]
+pub(crate) struct QueryArgs(CmnQueryArgs);
+impl QueryArgs {
+    /// Get a comma-separated list of query selection args.
+    /// todo: should be Display trait?
+    pub(super) fn query(&self) -> String {
+        Self::query_args(&self.0)
+    }
+    /// Get a comma-separated list of query selection args.
+    pub(super) fn query_args(args: &CmnQueryArgs) -> String {
+        let mut select = String::new();
+        if let Some(vg_name) = &args.name {
+            select.push_str(&format!("vg_name={vg_name},"));
+        }
+        if let Some(vg_uuid) = &args.uuid {
+            select.push_str(&format!("vg_uuid={vg_uuid},"));
+        }
+        if let Some(vg_tag) = &args.tag {
+            select.push_str(&format!("vg_tags={vg_tag},"));
+        }
+        select
+    }
+}
+impl From<CmnQueryArgs> for QueryArgs {
+    fn from(value: CmnQueryArgs) -> Self {
+        Self(value)
+    }
+}
+
+/// Used to decode the json output for vgs command to get
+/// the capacity and free size of a given volume group
+/// sudo vgs --options=vg_size,vg_free --units=b --nosuffix --reportformat=json
+///   {
+///       "report": [
+///           {
+///               "vg": [
+///                   {"vg_name": "pool", "vg_size":"15372124160",
+/// "vg_free":"15372124160"}                ]
+///           }
+///       ]
+///   }
+#[derive(Debug, Deserialize)]
+struct VolGroups {
+    /// Corresponds to the vg field in json output.
+    vg: Vec<VolumeGroup>,
+}
+
+/// An LVM Volume Group.
+#[derive(Debug, Clone, Deserialize)]
+pub struct VolumeGroup {
+    /// Corresponds to the vg_name field in json output, the name of the
+    /// volume group.
+    #[serde(rename = "vg_name")]
+    name: String,
+    /// Corresponds to the vg_uuid field in json output, the uuid of the
+    /// volume group.
+    #[serde(rename = "vg_uuid")]
+    uuid: String,
+    /// Corresponds to the vg_size field in json output, the total capacity of
+    /// volume group in bytes.
+    #[serde(deserialize_with = "de::number_from_string", rename = "vg_size")]
+    size: u64,
+    /// Corresponds to the vg_free field in json output, the free space on
+    /// volume group in bytes.
+    #[serde(deserialize_with = "de::number_from_string", rename = "vg_free")]
+    free: u64,
+    /// Corresponds to the vg_tags field in json output, the tags set in the
+    /// volume group.
+    #[serde(deserialize_with = "de::comma_separated", rename = "vg_tags")]
+    tags: Vec<Property>,
+    /// The physical vol disks used by the volume group.
```

Comment:
What do you mean by unpartitioned?

Reply:
With Mayastor, I guess we today the raw disk device to be provided(without any partitions) and the complete disk is used of creating a diskpool. Do we expect same here? In other words, can Mayastor use a logicalVolume created by collecting desired physical partitions from PVs?
Reference: https://en.wikipedia.org/wiki/Logical_Volume_Manager_(Linux)#/media/File:Lvm.svg

Reply:
You can use Mayastor with partitions AFAIK, not sure how much of that is tested but should work.
Yes it can, atm we're just given a VG, doesn't matter how the VG is composed

Reply:
We should enable both modes (Raw and partitioned). That offers the most flexibility to users.
note: A while ago, @tiagolobocastro  and I discussed that we should enable GPT awareness in Mayastor so that we can consume GPT partitions instead of the full raw disk. - This adds a level of safety to the system where we can label a GPT partition as a Mayastor disk entity and it wont get grabbed by something by mistake because it has a GPT label, - I think that was the plan.

Reply:
Yeah I have spoken with @dsavitskiy about this and it should be possible, though we need to support existing non-gpt pools as well.
For LVM backends this is probably not necessary as disks will already be taken by LVM itself.